addon/components/base/bs-form/element.js

```diff
@@ -707,25 +707,59 @@ export default FormGroup.extend({
 
     assert('showValidationOn must be a String or an Array', isArray(showValidationOn) || typeOf(showValidationOn) === 'string');
     if (isArray(showValidationOn)) {
-      return showValidationOn;
+      return showValidationOn.map((type) => {
+        return type.toLowerCase();
+      });
     }
 
     if (typeof showValidationOn.toString === 'function') {
-      return [showValidationOn];
+      return [showValidationOn.toLowerCase()];
     }
     return [];
   }),
 
   /**
    * @method showValidationOnHandler
+   * @params {Event} event
    * @private
    */
-  showValidationOnHandler(event) {
-    if (this.get('_showValidationOn').indexOf(event) !== -1) {
-      this.set('showOwnValidation', true);
+  showValidationOnHandler({ target, type }) {
+    // Should not do anything if
+    if (
+      // validations are already shown or
+      this.get('showOwnValidation') ||
+      // validations should not be shown for this event type or
+      this.get('_showValidationOn').indexOf(type) === -1 ||
+      // validation should not be shown for this event target
+      (
+        isArray(this.get('doNotShowValidationForEventTargets')) &&
+        this.get('doNotShowValidationForEventTargets.length') > 0 &&
+        [...this.element.querySelectorAll(this.get('doNotShowValidationForEventTargets').join(','))]
+          .some((el) => el.contains(target))
```

Comment:
An element contains itself. A little bit confusing but therefore we don't need to check `el === target`.